www/apps/book/app/debugging-and-testing/testing-tools/integration-tests/api-routes/page.mdx

```diff
@@ -0,0 +1,284 @@
+  req: MedusaRequest,
+  res: MedusaResponse
+) {
+  const helloModuleService: HelloModuleService = req.scope.resolve(
+    "helloModuleService"
+  )
+
+  const myCustom = await helloModuleService.createMyCustoms(
+    req.body
+  )
+
+  res.json({
+    my_custom: myCustom
+  })
+}
+```
+
+This API route creates a new record of `MyCustom`.
+
+To write tests for this API route, add the following at the end of the `testSuite` function in `integration-tests/http/custom-routes.spec.ts`:
+
+export const postHighlights = [
+  ["14", "api.post", "Send a POST request to the `/store/custom` API route."]
+]
+
+```ts title="integration-tests/http/custom-routes.spec.ts" highlights={postHighlights}
+// other imports...
+import HelloModuleService from "../../src/modules/hello/service"
+
+medusaIntegrationTestRunner({
+  testSuite: ({ api, getContainer }) => {
+    describe("Custom endpoints", () => {
+      // other tests...
+
+      describe("POST /store/custom", () => {
+        const id = "1"
+
+        it("Creates my custom", async () => {
+  
+          const response = await api.post(
+            `/store/custom`,
+            {
+              id,
+              name: "Test"
+            }
+          )
+  
+          expect(response.status).toEqual(200)
+          expect(response.data).toHaveProperty("my_custom")
+          expect(response.data.my_custom).toEqual({
+            id,
+            name: "Test",
+            created_at: expect.any(String),
+            updated_at: expect.any(String),
+          })
+        })
+      })
+    })
+  }
+})
+```
+
+This adds a test for the `POST /store/custom` API route. It uses `api.post` to send the POST request. The `api.post` method accepts as a second parameter the data to pass in the request body.
+
+The test passes if the response has:
+
+- Status code `200`.
+- A `my_custom` property in its data.
+- Its `id` and `name` match the ones provided to the request.
+
+### Tear Down Created Record
+
+To ensure consistency in the database for the rest of the tests after the above test is executed, utilize [Jest's setup and teardown hooks](https://jestjs.io/docs/setup-teardown) to delete the created record.
+
+Use the `getContainer` function passed as a parameter to the `testSuite` function to resolve a service and use it for setup or teardown purposes
+
+So, add an `afterAll` hook in the `describe` block for `POST /store/custom`:
+
+```ts title="integration-tests/http/custom-routes.spec.ts"
+// other imports...
+import HelloModuleService from "../../src/modules/hello/service"
+
+medusaIntegrationTestRunner({
+  testSuite: ({ api, getContainer }) => {
+    describe("Custom endpoints", () => {
+      // other tests...
+
+      describe("POST /store/custom", () => {
+        // ...
+        afterAll(() => async () => {
+          const helloModuleService: HelloModuleService = getContainer().resolve(
+            "helloModuleService"
+          )
+
+          await helloModuleService.deleteMyCustoms(id)
+        })
```

Comment:
question: Is this needed?

Reply:
Not needed per se, but considering this is an example guide I think it should provide a realistic approach of writing the tests.